### ISSUE TITLE
iOS

### ISSUE THREAD

**Issue by alihs** [geddski/csstyle#56]
The styles generated by cssstyle.io are not rendering on iOS. Can you confirm this please?
I need help.

**Comment by geddski** [comment]
Do you mean in mobile safari? Also if you could paste a small sample that
would help me debug the issue. It's likely that your HTML is just missing
the right classes.

**Comment by alihs** [comment]
<div class=“banner-box --bg-green”></div>

any styles on --bg-green doesnot come through on iOS safari and Windows Phone native browser

**Comment by geddski** [comment]
can you share your banner-box component (and bg-green option)?

**Comment by alihs** [comment]
```
<body id="EXAMPLE">
    
    <div class="banner-box --bg-green">
        This is a banner with a green background.
    </div>
    
    <div class="banner-box --bg-blue">
        This is a banner with a blue background.
    </div>
    
</body>
```

```
@include component(banner-box){
    padding: 30px;
    color: white;
    background-color: red;
    margin-bottom: 30px;
    text-align: center;
    
    @include option(bg-green){
        background-color: green;
    }
    
    @include option(bg-blue){
        background-color: blue;
    }
}
```


it is a very basic example. none of the option classes that are not rendering on IOS Safari and Micorosft Windows Phone

**Comment by geddski** [comment]
hmm, that should work just fine. Does it work on desktop chrome?

**Comment by geddski** [comment]
It's also possible that your sass hasn't run again since adding those options. Take a look at the compiled CSS file and see if it has them.

**Comment by alihs** [comment]
Works fine on desktop and compiles fine as well. Is it working fine on your iOS?

Ali Hussain

-----Original Message-----

**Comment by geddski** [comment]
yeah works just fine. csstyle.io is built with it, looks great on mobile. Maybe your CSS is being cached on mobile safari?

**Comment by alihs** [comment]
totally strange. my desktop site works fine but the mobile site is acting weird. Could it be a utf-8 issues?

**Comment by geddski** [comment]
I think I figured out the cause of the issue. I believe csstyle is missing an escape somewhere. In Chrome if your CSS has a `.--card` class, it will render just fine. On iOS it won't, unless you escape it properly: `\.--card`. I'll try to track down where this might be happening.

**Comment by alihs** [comment]
Thanks.
Also i too forgot to update, it was my css compressing engine that was
doing this.
I had to add a few rules to it and it worked.




- Ali Hussain
